src/internal/streamstore/streamstore.go

```diff
@@ -128,20 +129,20 @@ func (ss *streamStore) ReadBackupDetails(
 	for {
 		select {
 		case <-ctx.Done():
-			return nil, errors.New("context cancelled waiting for backup details data")
+			return nil, clues.New("context cancelled waiting for backup details data").WithClues(ctx)
```

Comment:
these are reported as errors (return-value wise) so we should be associating stack traces with them. I'm not sure off the top of my head if `clues.New` has that functionality yet

Reply:
We'll get there.

Reply:
tbh I'd prefer to continue having stack traces associated with log messages even if it requires us to do something like `clues.New(errors.New(...))` in the interim. Stack traces have been very helpful in tracking down exact locations of things and I don't want to lose them while we build out functionality